packages/relay/src/services/ReportService.ts

```diff
@@ -280,6 +302,290 @@ export class ReportService {
+            score: change,
+            type: repType,
+            reportId,
+        })
+    }
+
+    private isReportCompleted(report: ReportHistory): boolean {
+        const allAdjudicatorsClaimed =
+            report.adjudicatorsNullifier?.every((adj) => adj.claimed) ?? true
+        return (
+            !!report.reportorClaimedRep &&
+            !!report.respondentClaimedRep &&
+            allAdjudicatorsClaimed
+        )
+    }
+
+    getReputationType(
+        isPassed: boolean,
+        repUserType: RepUserType
+    ): ReputationType {
+        switch (repUserType) {
+            case RepUserType.VOTER:
+                return ReputationType.ADJUDICATE
+            case RepUserType.REPORTER:
+                return isPassed
+                    ? ReputationType.REPORT_SUCCESS
+                    : ReputationType.REPORT_FAILURE
+            case RepUserType.POSTER:
+                return ReputationType.BE_REPORTED
+            default:
+                throw new Error('Invalid RepUserType')
+        }
+    }
+
+    getClaimHelper(repUserType: RepUserType): ClaimHelpers {
+        switch (repUserType) {
+            case RepUserType.REPORTER:
+            case RepUserType.POSTER:
+                return ClaimHelpers.ReportNonNullifierVHelper
+            case RepUserType.VOTER:
+                return ClaimHelpers.ReportNullifierVHelper
+            default:
+                throw new Error('Invalid RepUserType')
+        }
+    }
+
+    checkAdjudicatorNullifier(report: ReportHistory, nullifier: string) {
+        if (
+            !report.adjudicatorsNullifier?.some(
+                (adj) => adj.nullifier === nullifier
+            )
+        ) {
+            throw new Error('Invalid adjudicator nullifier')
+        }
+    }
+
+    checkAdjudicatorIsClaimed(report: ReportHistory, nullifier: string) {
+        if (
+            report.adjudicatorsNullifier?.some(
+                (adj) => adj.nullifier === nullifier && adj.claimed
+            )
+        ) {
+            throw UserAlreadyClaimedError
+        }
+    }
+
+    checkRespondentEpochKey(report: ReportHistory, epochKey: string) {
+        if (report.respondentEpochKey !== epochKey) {
+            throw new Error('Invalid respondent epoch key')
+        }
+    }
+
+    checkRespondentIsClaimed(report: ReportHistory) {
+        if (report.respondentClaimedRep) {
+            throw UserAlreadyClaimedError
+        }
+    }
+
+    checkReporterIsClaimed(report: ReportHistory) {
+        if (report.reportorClaimedRep) {
+            throw UserAlreadyClaimedError
+        }
+    }
+
+    getClaimMethod(repUserType: RepUserType, isPassed: boolean): ClaimMethods {
+        if (isPassed) {
+            return ClaimMethods.CLAIM_POSITIVE_REP
+        } else {
+            if (repUserType === RepUserType.POSTER) {
+                return ClaimMethods.CLAIM_NEGATIVE_REP
+            }
+            return ClaimMethods.CLAIM_NEGATIVE_REP
+        }
+    }
```

Comment:
The implementation of this function should be below

```ts
getClaimMethod(repUserType: RepUserType, isPassed: boolean): ClaimMethods | null {
    switch (repUserType) {
        case RepUserType.POSTER:
            return isPassed ? ClaimMethods.CLAIM_NEGATIVE_REP : null
        case RepUserType.REPORTER:
            return isPassed
                ? ClaimMethods.CLAIM_POSITIVE_REP
                : ClaimMethods.CLAIM_NEGATIVE_REP
        case RepUserType.VOTER:
            return ClaimMethods.CLAIM_POSITIVE_REP
        default:
            throw new Error('Invalid RepUserType')
    }
}
```

Reply:
fixed